### ISSUE TITLE
generated CTD missing required ctdVersion field

### ISSUE THREAD

**Issue by mr-c** [seqan/seqan#2280]
https://github.com/WorkflowConversion/CTDSchema/blob/master/CTD.xsd#L64

See also https://github.com/WorkflowConversion/CTDConverter/issues/35

  - [ ] ctdVersion not written
  - [ ] insegt: `UserWarning: Restriction [gff] of a single value found for parameter [fusion-output].`
  - [ ] flexbar: `Invalid default value(s) provided for parameter barcode-min-overlap of type <type 'int'>: ''`

**Comment by mr-c** [comment]
```
INFO: Parsing insegt.ctd          
/home/mcrusoe/miniconda2/envs/ctd-converter/lib/python2.7/site-packages/CTDopts/CTDopts.py:156: UserWarning: Restriction [gff] of a single value found
for parameter [fusion-output].               
Restrictions should be comma separated value lists or colon separated values to indicate numeric ranges (e.g., 'true,false', '0:14', '1:', ':2.8')
Will use a restriction with one possible value of choice.                               
  (attribs['restrictions'], attribs['name'])) 
```

**Comment by h-2** [comment]
Assigning @smehringer as the person working on the arg parser.

@mr-c: Could you create a check-list with issues you found? I have already started with it :)

**Comment by h-2** [comment]
@mr-c In general I am not too happy with the CTDs, but we will still support them in SeqAn3. If, however, you have a superior standardised format from the CWL project we would be very open to contributions, though! Or we could maybe have a student implement another format at some point...

**Comment by mr-c** [comment]
@h-2 People seem pretty happy with http://www.commonwl.org/v1.0/CommandLineTool.html :-)
V1.0.x doesn't include value restrictions, but they could be developed as an extension which would inform a future version.

In fact I met an OpenMS / CTD person two days ago at the ELIXIR Tools platform meeting; between him and others I expect there will be a proposal for extending CWL for value restrictions soon. Seqan participation would be very welcome.

TL;DR: someone with C++ skills should be able to add CWL export. For example, here's the Python argparse equivalent: https://github.com/erasche/argparse2tool

**Comment by rrahn** [comment]
@mr-c I am in contact with the OpenMS guys. Probably we'll sketch and implement it together.
However, the ctdversion issues seems to be in SeqAn since the beginning 😭 
That we can fix easily for 2.4.1. Regarding flexbar, this is an external project, but we will have a look.

**Comment by mr-c** [comment]
@rrahn Thank you for the update!

**Comment by smehringer** [comment]
@rrahn Which ctdVersion do we support?

**Comment by h-2** [comment]
fixed
